### PR TITLE
Enable change language

### PR DESCRIPTION
**Pull request by ZeTheGreat**
Just let the user change the the language based on arguments to loader

### CODE REVIEW

**Comment by kamilglod** [comment]
@ZeTheGreat thanks for your contribution, however this repo is very old and very outdated. Maybe the best way would be to archive it.

When it comes to your proposal: I'm not super up to date with the i18next stuff, but what's the point of setting this language at all? I think we need to pass language dict with translations as described in README so the language settings in i18next should be unnecessary.

**Comment by ZeTheGreat** [comment]
Hello!

Thank you for your attention!

Your repo works pretty well! And it is recommended in i18next documentation, I think it is a good idea to not archive it!

The point of this language is that I have multiple (for each language) webpack configs in the same build, this approach makes my bundle share the same i18next instance, so if I use `init` the language of all bundles will be just the last one!

So I create one i18next with all languages and between transpile I change the language for the correct one!

**Comment by ZeTheGreat** [comment]
https://www.i18next.com/overview/plugins-and-utils

take a look! (Just search for your library name!)

**Comment by ZeTheGreat** [comment]
for example:
Webpack config output
```
[
     {lng: 'en-US' ...},
     {lng: 'es-US '}
]
```


When the webpack will create the bundle he will use the same instance of i18n for both configs...

This is why I need to change the language 


PS: I added the on the READ.me about the new config

**Comment by ZeTheGreat** [comment]
I tested it on my code and work fine with and without the config

What do you think?

**Comment by kamilglod** [comment]
@ZeTheGreat can you post here an example full webpack config? I'm not sure but can't you achive the same just by manually changing language based on the settings before passing it to the plugin config? Something like:
```js
var i18next = require('i18next');

const languages = [
     {lng: 'en-US' ...},
     {lng: 'es-US '}
];
const language = languages.find(({lng}) => lng === "en-US");

i18next.init({
  ...,
  resources: language,
});

```

**Comment by ZeTheGreat** [comment]
Yes! This is the helper that create the base of webpack config:

```js
...
// IMPORT THE i18next
const i18next = require('i18next')
...

const createConfigCountry = (env, country) => {
  ...
  const webpackConfig = {
      ...
    },
    resolve: {
      extensions: ['.tsx', '.ts', '.js'],
      plugins: [new TsconfigPathsPlugin()]
    }
  }

  ...

  const { languageConfig, countryCode } = config

  console.info(`Building ${countryCode}`)
  // INSTANCE OF i18N
  i18next.init({
    debug: true
  })
  for (const language in languageConfig) {
    // IMPORT LANGUAGE
    const lowerLanguage = language.toLowerCase()
    const translationList = JSON.parse(getFile(`./src/i18n/${lowerLanguage}.json`))
    config.activeLanguage = language
    // ADDING LANGUAGE
    i18next.addResourceBundle(language, 'translation', translationList, true, true)

    ...

    // PREPARING PAGE WE CALL IT JOURNEY

    for (const journeyName in journeys) {
      ...

      const PageConfig = {
        title: journeyName,
        filename: `${journeyName}.html`,
        template: pathBasePage,
        templateParameters: journey.config,
        inject: 'body',
        scriptLoading: 'defer',
        favicon: './src/data/img/icon.ico',
        meta,
        hash: true,
        cache: true,
        showErrors: true,
        chunks: [journeyName]
      }

      entry[journeyName] = journey.sourcePathJs
      // ADD THE PAGES INTO CONFIG
      journeyTemplates.push(new HtmlWebpackPlugin(PageConfig))
    }

    const outputDir = `dist/${country}/${language}`
    const sourcePath = path.join(__dirname, 'src')

    // PREPARE THE WEBPACK CONFIG
    const languageWebpackConfig = {
      module: {
        rules: [
          // TS
          {
            test: /\.tsx?$/,
            use: ['ts-loader', {
              loader: 'i18next-loader',
              options: {
                lng: language // Here i add the lng
              }
            }],
            exclude: [/node_modules/]
          }
        ]
      },
      name: language,
      entry,
      plugins: [
        ...journeyTemplates,
        new webpack.DefinePlugin({
          ABIENV: `'${JSON.stringify({ country })}'`
        }),
        new CleanWebpackPlugin(),
        new MiniCssExtractPlugin({
          filename: 'css/[name]-[contenthash].css'
        }),
        new PurgeCSSPlugin({
          paths: glob.sync(`${sourcePath}/**/*`, { nodir: true })
        })
      ],
      output: {
        filename: 'js/[name]-[contenthash].js',
        path: path.resolve(__dirname, outputDir)
      }
    }
    countryWebpackConfigArray.push(merge(webpackConfig, languageWebpackConfig))
  }
  // EXPORT ARRAY
  return countryWebpackConfigArray
}

module.exports = createConfigCountry

```

**Comment by ZeTheGreat** [comment]
And here we have the dev config for example:

```js
const { merge } = require('webpack-merge')
// HERE WE HAVE THE HELPER THAT I SEND BEFORE
const webpackBase = require('./webpack.base.js')

module.exports = ({ country, environment }) => {
  const HOST_PROTOCOL = 'http'
  const HOST_NAME = 'localhost'
  const HOST_PORT = 8080
  const BASE_PATH = '/'
  const URL = `${HOST_PROTOCOL}://${HOST_NAME}:${HOST_PORT}${BASE_PATH}`

  console.info('Local Host:', URL)
  console.info('Country:', country)
  console.info('Environment:', environment)

  const common = webpackBase(environment, country)

  const webpackConfigs = []
  common.forEach(function (webpackConfig, index) {
    console.info(webpackConfig.name)
    webpackConfigs[index] = merge(webpackConfig, {
      mode: 'development',
      devtool: 'inline-source-map',
      // entry: {
      //   serverMock: path.resolve(__dirname, 'serverMockClass/serverMock.js')
      // },
      output: {
        publicPath: URL + webpackConfig.name
      }
    })
  })

  return webpackConfigs
}

```

**Comment by kamilglod** [comment]
Thanks, please explain me why you cannot call this `i18next.changeLanguage(language)` inside of your `for (const language in languageConfig)`  loop?

**Comment by ZeTheGreat** [comment]
Because this time I am just creating the config object, this code is already ran when the server run

File: README.md [hunk 1]
```diff
@@ -59,6 +59,7 @@ this is a webpack loader that can translate your code and generate bundle per ea
 ### query params:
 - `funcName`: change default function name (default is `__()`)
 - `quotes`: choose how to generates translated strings in code (default is `''`)
+- `lng`: choose the language of which bundle (default is undefined, if we pass nothing will use on the init)
```
Comment (kamilglod) [inline README.md]: I don't quite understand this sentence. Please change it to something like `set i18next language (useful when having webpack config per language)`

**Review by kamilglod — APPROVED** [review]
I think I understand it a little more, thanks for your patient @ZeTheGreat

File: README.md (hunk 1 above)
Comment (ZeTheGreat) [reply README.md]: Done my friend!

**Comment by ZeTheGreat** [comment]
Thank you for your time :)

**Pull request merged**